runtime/wasm/src/module/mod.rs

```diff
@@ -207,23 +210,43 @@ where
     ) -> Result<Vec<EntityOperation>, FailureError> {
         self.start_time = Instant::now();
 
+        let block = self.ctx.block.block.clone();
+        let transaction = self.ctx.transaction.clone();
+
         // Prepare an EthereumEvent for the WASM runtime
-        let event = EthereumEventData {
-            block: EthereumBlockData::from(&self.ctx.block.block),
-            transaction: EthereumTransactionData::from(self.ctx.transaction.deref()),
-            address: log.address,
-            log_index: log.log_index.unwrap_or(U256::zero()),
-            transaction_log_index: log.transaction_log_index.unwrap_or(U256::zero()),
-            log_type: log.log_type.clone(),
-            params,
+        // Decide on the destination type using the version provided
+        // in the subgraph manifest (SpecVersion)
+        let event = if self.host_exports.spec_version > Version::parse("0.0.1").unwrap() {
```

Comment:
imo it's more meaningful to check for `>= 0.0.2` than `> 0.0.1`, since then we know the minimum supported version.

To avoid the unwrap, you can use `Version::new(0, 0, 1)`.